crates/rustc_codegen_spirv/build.rs

```diff
@@ -63,21 +73,189 @@ fn check_toolchain_version() -> Result<(), Box<dyn Error>> {
                 .reduce(|a, b| a + "\n" + &b)
                 .unwrap_or_default();
 
-            return Err(Box::<dyn Error>::from(format!(
+            return Err(format!(
                 r#"error: wrong toolchain detected (found commit hash `{current_hash}`, expected `{required_hash}`).
 Make sure your `rust-toolchain.toml` file contains the following:
 -------------
 {stripped_toolchain}
 -------------"#
-            )));
+            ).into());
         }
     }
 
     Ok(())
 }
 
+/// Copy `rustc_codegen_ssa` (from the `rustc-dev` component) and patch it to
+/// produce a "pqp" ("pre-`qptr`-patched") version that maintains compatibility
+/// with "legacy" Rust-GPU pointer handling (mainly typed `alloca`s).
+//
+// FIXME(eddyb) get rid of this as soon as it's not needed anymore.
```

Comment:
We should have a task written for this so we don't lose track

Reply:
- https://github.com/Rust-GPU/rust-gpu/issues/182